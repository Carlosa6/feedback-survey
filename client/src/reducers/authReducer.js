import { FETCH_USER, FETCH_USER_SUCCEESS } from '../actions/types';

export const initialState = {
    auth: {},
    loaded: false
};

export const authReducer = function (state = null, action) {
    switch (action.type) {
        case FETCH_USER:
            return action.payload || false;
        case FETCH_USER_SUCCEESS:
            return { payload: action.payload, loaded: true }
        default:
            return state;
    }
}

export default authReducer;