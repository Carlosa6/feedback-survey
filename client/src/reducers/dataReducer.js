import { FETCH_SURVEY } from '../actions/types';

export const initialState = {
    surveys: {},
    loaded: false
};

export const dataReducer = function (state = null, action) {
    switch (action.type) {
        case FETCH_SURVEY: {
            return action.payload || false;
        }
        default:
            return state;
    }
}

export default dataReducer;