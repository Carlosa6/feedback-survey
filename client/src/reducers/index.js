import { combineReducers } from 'redux';
import { reducer as reduxForm } from 'redux-form';
import authReducer from "./authReducer";
import dataReducer from './dataReducer';
// import dataReducer from "./dataReducer";

export default combineReducers({
    auth: authReducer,
    form: reduxForm,
    surveys: dataReducer
});