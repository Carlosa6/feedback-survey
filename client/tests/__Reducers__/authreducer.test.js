import authReducer from '../../src/reducers/authReducer'
import { initialState } from '../../src/reducers/authReducer'

describe('Auth reducer', () => {
    it('should return the initial state', () => {
        expect(authReducer(undefined, {})).toMatchSnapshot()
    })

    it('should handle FETCH_USER', () => {
        expect(
            authReducer(initialState,
                {
                    type: 'FETCH_USER'
                })
        ).toMatchSnapshot()
    })
})