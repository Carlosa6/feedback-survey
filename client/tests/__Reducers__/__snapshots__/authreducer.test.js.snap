// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth reducer should handle FETCH_USER 1`] = `
Object {
  "auth": Object {},
  "loaded": false,
}
`;

exports[`Auth reducer should return the initial state 1`] = `null`;
